tests/core/test_actions.py

```diff
@@ -2825,3 +2825,72 @@ async def test_action_extract_slots_priority_of_slot_mappings():
     )
     tracker.update_with_events(events, domain=domain)
     assert tracker.get_slot("location_slot") == entity_value
+
+
+async def test_action_extract_slots_does_allows_slotset_for_same_value(
+    caplog: LogCaptureFixture,
+):
+    domain_yaml = textwrap.dedent(
+        f"""
+        version: "{LATEST_TRAINING_DATA_FORMAT_VERSION}"
+
+        slots:
+          custom_slot_a:
+            type: text
+            influence_conversation: false
+            mappings:
+            - type: custom
+              action: custom_extract_action
+
+        actions:
+        - custom_extract_action
+        - action_validate_slot_mappings
+        """
+    )
+    domain = Domain.from_yaml(domain_yaml)
+    event = UserUttered("Hi")
+    tracker = DialogueStateTracker.from_events(
+        sender_id="test_id", evts=[event], slots=domain.slots
```

Comment:
This unit test you added fails in the CI because of a misunderstanding on how the action_extract_slots runs (this runs on every user turn, so the agent would predict and run the custom action only once, not twice as in your example). I'd remove the 2nd call of the custom action because that's not plausible, and instead add a SlotSet event in the tracker , in the evts parameter (this is to mimick the ticket scenario where the slot had already been set to a value). Then the assertion should be that the duplicate SlotSet event was added to the tracker.